js/client-battle.js

```diff
@@ -396,14 +396,15 @@
 						if (move.id === 'Struggle' || move.id === 'Recharge') pp = '&ndash;';
 						if (move.id === 'Recharge') move.type = '&ndash;';
 						if (name.substr(0, 12) === 'Hidden Power') name = 'Hidden Power';
+						var moveType = this.getMoveType(move, switchables[pos]);
```

Comment:
I like two functions for now.

PP is already sent from the server so it's always accurate.